File: backend/app/schemas/issues.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class IssueAuthor(BaseModel):
    login: str
    avatar_url: str


class Issue(BaseModel):
    number: int
    title: str
    state: str  # "open" | "closed"
    author: IssueAuthor
    labels: list[str]
    comments: int
    html_url: str
    created_at: datetime
    closed_at: datetime | None = None


class IssuesSummary(BaseModel):
    open_count: int
    closed_count: int
    total_count: int
    issues: list[Issue]

File: frontend/types/issues.ts
export interface IssueAuthor {
  login: string;
  avatar_url: string;
}

export interface Issue {
  number: number;
  title: string;
  state: "open" | "closed";
  author: IssueAuthor;
  labels: string[];
  comments: number;
  html_url: string;
  created_at: string;
  closed_at: string | null;
}

export interface IssuesSummary {
  open_count: number;
  closed_count: number;
  total_count: number;
  issues: Issue[];
}
